src/main/java/hudson/plugins/active_directory/CacheConfiguration.java

```diff
@@ -0,0 +1,70 @@
+package hudson.plugins.active_directory;
+
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Cache for Users and Groups
+ *
+ */
+public class CacheConfiguration<K,V,E extends Exception> {
+    private final int size;
+    private final int ttl;
+
+    @DataBoundConstructor
+    public CacheConfiguration(int size, int ttl) {
+        this.size = Math.max(10, Math.min(size, 1000));
+        this.ttl = Math.max(30, Math.min(ttl, 3600));
+    }
+
+    public int getSize() {
+        return size;
+    }
+
+    public int getTtl() {
+        return ttl;
+    }
+
+    public static class CacheEntry<T> {
+        private final long expires;
+        private final T value;
+
+        public CacheEntry(int ttlSeconds, T value) {
+            this.expires = System.currentTimeMillis() + TimeUnit.SECONDS.toMillis(ttlSeconds);
+            this.value = value;
+        }
+
+        public T getValue() {
+            return value;
+        }
+
+        public boolean isValid() {
+            return System.currentTimeMillis() < expires;
+        }
+    }
+
+    /**
+     * While we could use Guava's CacheBuilder the method signature changes make using it problematic.
+     * Safer to roll our own and ensure compatibility across as wide a range of Jenkins versions as possible.
+     *
+     * @param <K> Key type
+     * @param <V> Cache entry type
+     */
+    public static class CacheMap<K, V> extends LinkedHashMap<K, CacheEntry<V>> {
+
+        private final int cacheSize;
+
+        public CacheMap(int cacheSize) {
+            super(cacheSize + 1); // prevent realloc when hitting cache size limit
+            this.cacheSize = cacheSize;
+        }
+
+        @Override
+        protected boolean removeEldestEntry(Map.Entry<K, CacheEntry<V>> eldest) {
```

Comment:
this does not ever seem to be called?

in which case just use a normal ConcurrentHashMap here and remove the syncronization